### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Kapidastra**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,29 @@
...
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateCopy) {
+        delete stateCopy[key];
+      }
+    }
+    result.push({ ...stateCopy });
+  }
```
Comment (anastasiia-tilikina) [inline src/transformStateWithClones.js:30]: let's use switch case here

**Review by anastasiia-tilikina — CHANGES_REQUESTED** [review]
Well done!
Just one thing to improve

**Review by Esceype — CHANGES_REQUESTED** [review]
I'm sorry for the double rejection, pay attention only to the previous comment.

**Review by VitaliyBondarenko1982 — APPROVED** [review]
GJ!

**Pull request closed without merging**